src/Microsoft.VisualStudio.Threading.Shared/AsyncBarrier.cs

```diff
@@ -25,24 +25,22 @@ public class AsyncBarrier
         private readonly int participantCount;
 
         /// <summary>
-        /// The number of participants that have not yet signaled the barrier.
-        /// </summary>
-        private int remainingParticipants;
-
-        /// <summary>
         /// The set of participants who have reached the barrier, with their awaiters that can resume those participants.
         /// </summary>
-        private ConcurrentStack<TaskCompletionSource<bool>> waiters;
+        private Stack<TaskCompletionSource<EmptyStruct>> waiters;
```

Comment:
Then you'd not need participantCount as well - the size of the array will tell you that.